pkg/controller/controller_utils.go

```diff
@@ -54,6 +54,13 @@ var (
 	KeyFunc = framework.DeletionHandlingMetaNamespaceKeyFunc
 )
 
+type ResyncPeriodFunc func() time.Duration
```

Comment:
I can't understand why you're injecting a function, but all the functions you use just return a constant. If you're doing it for the type protection from the compiler, why not just `type ResyncPeriod time.Duration`?